<H2> Capstone Project - The Battle of the Neighborhoods (Week 2)</H2>

<H3>Applied Data Science Capstone by IBM/Coursera</H3>

<B> Table of Contents </B>

* [Introduction: Business Problem](#introduction)
* [Data](#data)
* [Analysis](#analysis)
* [Results and Discussion](#discussion)
* [Conclusion](#conclusion)

<B><font color = 'blue'> Introduction: Business Problem </font><a name="introduction"></a> </B>

In this project we will compare the top 5 drinking locations in New York City (NYC) to Toronto. Specifically, this report will be targeted to stakeholders interested in drinking places in NYC versus Toronto to determine which city has better drinks.

Only the top 5 drinking places will be considered for each city. The data will be scraped from Foursquare and will be sort by popularity. Data to consider assessing will include price, rating and number of likes. Additional unique features may also be taken into consideration. <BR><BR>


<B><font color = 'blue'> Data </font><a name="data"></a> </B>

Based on our defined business problem, factors that will influence our decisions are:
* Price
* Rating
* Number of Likes
* Unique features if any
<BR>

In [7]:
#Import the needed libraries
import requests 
import pandas as pd 
import numpy as np 
import random 
!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim 

from IPython.display import Image 
from IPython.core.display import HTML 
    
from pandas.io.json import json_normalize

!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Solving environment: done

# All requested packages already installed.

Folium installed
Libraries imported.


In [12]:
CLIENT_ID = '1DMKQ5NQ5VMJLWX5GSWMA0HGJCXJMWBKY3GURRVXVD3MQMLW' 
CLIENT_SECRET = 'OFVFGV1NCWSYF5X5B0S4ZTJYE4GPK4FYOX2MYJ1AV1NAUNUU'
VERSION = '20180604'
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 1DMKQ5NQ5VMJLWX5GSWMA0HGJCXJMWBKY3GURRVXVD3MQMLW
CLIENT_SECRET:OFVFGV1NCWSYF5X5B0S4ZTJYE4GPK4FYOX2MYJ1AV1NAUNUU


In [8]:
geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode('New York City, NY')
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

40.7127281 -74.0060152


In [9]:
radius = 500

In [13]:
nyc_drinks_url = 'https://api.foursquare.com/v2/venues/explore?ll=40.7127281,-74.0060152&section=drinks&sortByPopularity=0&limit=5&client_id={}&client_secret={}&v={}&radius={}'.format(CLIENT_ID, CLIENT_SECRET, VERSION, radius)
nyc_drinks_result = requests.get(nyc_drinks_url).json()
nyc_drinks_result

{'meta': {'code': 200, 'requestId': '5e33813db57e88001bf54b50'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': '$-$$$$', 'key': 'price'},
    {'name': 'Open now', 'key': 'openNow'}]},
  'headerLocation': 'Downtown Manhattan',
  'headerFullLocation': 'Downtown Manhattan, New York',
  'headerLocationGranularity': 'neighborhood',
  'query': 'drinks',
  'totalResults': 41,
  'suggestedBounds': {'ne': {'lat': 40.7172281045, 'lng': -74.00008952063419},
   'sw': {'lat': 40.7082280955, 'lng': -74.0119408793658}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '57f0689d498e7d49d9189369',
       'name': 'The Bar Room at Temple Court',
       'location': {'address': '123 Nassau St',
        'lat': 40.7114477287544,
        'lng': -74.006801570

In [14]:
nyc_venue = nyc_drinks_result['response']['groups'][0]['items']
nyc_drinks_df = json_normalize(nyc_venue)
nyc_drinks_df

referralId  reasons.count  \
0  e-4-57f0689d498e7d49d9189369-0              0   
1  e-4-534c9d7b498e1bdd443a40e1-1              0   
2  e-4-4db5fbfe43a1369cb5dc2a1f-2              0   
3  e-4-4aa0669cf964a5207e3f20e3-3              0   
4  e-4-433b2e80f964a52036281fe3-4              0   

                                       reasons.items  \
0  [{'summary': 'This spot is popular', 'type': '...   
1  [{'summary': 'This spot is popular', 'type': '...   
2  [{'summary': 'This spot is popular', 'type': '...   
3  [{'summary': 'This spot is popular', 'type': '...   
4  [{'summary': 'This spot is popular', 'type': '...   

                   venue.id                    venue.name  \
0  57f0689d498e7d49d9189369  The Bar Room at Temple Court   
1  534c9d7b498e1bdd443a40e1                       Racines   
2  4db5fbfe43a1369cb5dc2a1f    Maxwell's Bar & Restaurant   
3  4aa0669cf964a5207e3f20e3                     The Wooly   
4  433b2e80f964a52036281fe3         FlashDancers Downtown   

  venue.location.address  venue.location.lat  venue.location.lng  \
0          123 Nassau St           40.711448          -74.006802   
1         94 Chambers St           40.714754          -74.007581   
2            59 Reade St           40.714944          -74.006485   
3          11 Barclay St           40.712212          -74.008357   
4           59 Murray St           40.714350          -74.009800   

                       venue.location.labeledLatLngs  venue.location.distance  \
0  [{'label': 'display', 'lat': 40.7114477287544,...                      157   
1  [{'label': 'display', 'lat': 40.71475415146123...                      261   
2  [{'label': 'display', 'lat': 40.71494392643706...                      249   
3  [{'label': 'display', 'lat': 40.71221245515355...                      205   
4  [{'label': 'display', 'lat': 40.71435, 'lng': ...                      366   

   ... venue.photos.count venue.photos.groups venue.location.crossStreet  \
0  ...                  0                  []                        NaN   
1  ...                  0                  []                        NaN   
2  ...                  0                  []                at Broadway   
3  ...                  0                  []                at Broadway   
4  ...                  0                  []  Church St & West Broadway   

  venue.delivery.id                                 venue.delivery.url  \
0               NaN                                                NaN   
1               NaN                                                NaN   
2            315611  https://www.seamless.com/menu/maxwells-59-read...   
3           1085252  https://www.grubhub.com/restaurant/the-wooly-p...   
4               NaN                                                NaN   

  venue.delivery.provider.name       venue.delivery.provider.icon.prefix  \
0                          NaN                                       NaN   
1                          NaN                                       NaN   
2                     seamless  https://fastly.4sqi.net/img/general/cap/   
3                      grubhub  https://fastly.4sqi.net/img/general/cap/   
4                          NaN                                       NaN   

  venue.delivery.provider.icon.sizes  \
0                                NaN   
1                                NaN   
2                           [40, 50]   
3                           [40, 50]   
4                                NaN   

          venue.delivery.provider.icon.name venue.venuePage.id  
0                                       NaN                NaN  
1                                       NaN                NaN  
2  /delivery_provider_seamless_20180129.png           45470725  
3   /delivery_provider_grubhub_20180129.png           82838739  
4                                       NaN           79597613  

[5 rows x 28 columns]

In [15]:
#overviewing the columns to see if there are unique features we can pull
list(nyc_drinks_df.columns) 

['referralId',
 'reasons.count',
 'reasons.items',
 'venue.id',
 'venue.name',
 'venue.location.address',
 'venue.location.lat',
 'venue.location.lng',
 'venue.location.labeledLatLngs',
 'venue.location.distance',
 'venue.location.postalCode',
 'venue.location.cc',
 'venue.location.neighborhood',
 'venue.location.city',
 'venue.location.state',
 'venue.location.country',
 'venue.location.formattedAddress',
 'venue.categories',
 'venue.photos.count',
 'venue.photos.groups',
 'venue.location.crossStreet',
 'venue.delivery.id',
 'venue.delivery.url',
 'venue.delivery.provider.name',
 'venue.delivery.provider.icon.prefix',
 'venue.delivery.provider.icon.sizes',
 'venue.delivery.provider.icon.name',
 'venue.venuePage.id']

In [16]:
nyc_drinks2_df = nyc_drinks_df[['venue.id','venue.name','venue.location.lat', 'venue.location.lng', 'venue.location.formattedAddress', 'venue.delivery.url']]
nyc_drinks2_df

venue.id                    venue.name  venue.location.lat  \
0  57f0689d498e7d49d9189369  The Bar Room at Temple Court           40.711448   
1  534c9d7b498e1bdd443a40e1                       Racines           40.714754   
2  4db5fbfe43a1369cb5dc2a1f    Maxwell's Bar & Restaurant           40.714944   
3  4aa0669cf964a5207e3f20e3                     The Wooly           40.712212   
4  433b2e80f964a52036281fe3         FlashDancers Downtown           40.714350   

   venue.location.lng                    venue.location.formattedAddress  \
0          -74.006802  [123 Nassau St, New York, NY 10038, United Sta...   
1          -74.007581  [94 Chambers St, New York, NY 10007, United St...   
2          -74.006485  [59 Reade St (at Broadway), New York, NY 10007...   
3          -74.008357  [11 Barclay St (at Broadway), New York, NY 102...   
4          -74.009800  [59 Murray St (Church St & West Broadway), New...   

                                  venue.delivery.url  
0                                                NaN  
1                                                NaN  
2  https://www.seamless.com/menu/maxwells-59-read...  
3  https://www.grubhub.com/restaurant/the-wooly-p...  
4                                                NaN

In [17]:
nyc_venues_id_list = nyc_drinks_df['venue.id'].values.tolist()
nyc_venues_id_list

['57f0689d498e7d49d9189369',
 '534c9d7b498e1bdd443a40e1',
 '4db5fbfe43a1369cb5dc2a1f',
 '4aa0669cf964a5207e3f20e3',
 '433b2e80f964a52036281fe3']

In [18]:
nyc_rating = []
nyc_price = []
nyc_likes = []

for nyc_venues1 in nyc_venues_id_list:
    url2 = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(nyc_venues1, CLIENT_ID, CLIENT_SECRET, VERSION)
    result2 = requests.get(url2).json()
    result4 = result2['response']['venue']['rating']
    result5 = result2['response']['venue']['attributes']['groups'][0]['summary']
    result6 = result2['response']['venue']['likes']['count']
    nyc_rating.append(result4)
    nyc_price.append(result5)
    nyc_likes.append(result6)

nyc_rpl = pd.DataFrame({'Price': nyc_price, 'Rating': nyc_rating, 'Number of likes': nyc_likes})
nyc_rpl

Price  Rating  Number of likes
0   $$$     9.4              182
1   $$$     8.6              115
2    $$     8.1              165
3   $$$     7.8              173
4   $$$     8.7               87

In [19]:
nyc_v1 = pd.concat([nyc_drinks2_df, nyc_rpl], axis=1)
nyc_v1

venue.id                    venue.name  venue.location.lat  \
0  57f0689d498e7d49d9189369  The Bar Room at Temple Court           40.711448   
1  534c9d7b498e1bdd443a40e1                       Racines           40.714754   
2  4db5fbfe43a1369cb5dc2a1f    Maxwell's Bar & Restaurant           40.714944   
3  4aa0669cf964a5207e3f20e3                     The Wooly           40.712212   
4  433b2e80f964a52036281fe3         FlashDancers Downtown           40.714350   

   venue.location.lng                    venue.location.formattedAddress  \
0          -74.006802  [123 Nassau St, New York, NY 10038, United Sta...   
1          -74.007581  [94 Chambers St, New York, NY 10007, United St...   
2          -74.006485  [59 Reade St (at Broadway), New York, NY 10007...   
3          -74.008357  [11 Barclay St (at Broadway), New York, NY 102...   
4          -74.009800  [59 Murray St (Church St & West Broadway), New...   

                                  venue.delivery.url Price  Rating  \
0                                                NaN   $$$     9.4   
1                                                NaN   $$$     8.6   
2  https://www.seamless.com/menu/maxwells-59-read...    $$     8.1   
3  https://www.grubhub.com/restaurant/the-wooly-p...   $$$     7.8   
4                                                NaN   $$$     8.7   

   Number of likes  
0              182  
1              115  
2              165  
3              173  
4               87

In [20]:
#renaming columns and replacing NaN values
nyc_v2 = nyc_v1.rename(columns={"venue.id": "ID", "venue.name": "Venue", "venue.location.lat": "Latitude", "venue.location.lng": "Longitude", "venue.location.formattedAddress": "Address", "venue.delivery.url": "Delivery url"})
nyc_v2.fillna("Not available", inplace = True)

In [22]:
#price range per Foursquare
nyc_v3 = nyc_v2.replace({'$$': '$10-$20', '$$$':'$20-$30'})

In [23]:
geolocator2 = Nominatim(user_agent="foursquare_agent")
location2 = geolocator2.geocode('Toronto, ON')
latitude2 = location2.latitude
longitude2 = location2.longitude
print(latitude2, longitude2)

43.653963 -79.387207


In [24]:
toronto_drinks_url = 'https://api.foursquare.com/v2/venues/explore?ll=43.653963,-79.387207&section=drinks&sortByPopularity=0&limit=5&client_id={}&client_secret={}&v={}&radius={}'.format(CLIENT_ID, CLIENT_SECRET, VERSION, radius)
toronto_drinks_result = requests.get(toronto_drinks_url).json()
toronto_drinks_result

{'meta': {'code': 200, 'requestId': '5e3382320de0d9001b1aaa1a'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': 'Open now', 'key': 'openNow'}]},
  'headerLocation': 'Bay Street Corridor',
  'headerFullLocation': 'Bay Street Corridor, Toronto',
  'headerLocationGranularity': 'neighborhood',
  'query': 'drinks',
  'totalResults': 13,
  'suggestedBounds': {'ne': {'lat': 43.6584630045, 'lng': -79.38099903084075},
   'sw': {'lat': 43.649462995499995, 'lng': -79.39341496915925}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '4ad7bf71f964a5205b0e21e3',
       'name': 'Sin and Redemption',
       'location': {'address': '136 McCaul St.',
        'crossStreet': 'at Dundas St.',
        'lat': 43.65466135819523,
        'lng': -79.3917372822

In [25]:
toronto_venue = toronto_drinks_result['response']['groups'][0]['items']
toronto_drinks_df = json_normalize(toronto_venue)

toronto_drinks_df

referralId  reasons.count  \
0  e-4-4ad7bf71f964a5205b0e21e3-0              0   
1  e-4-4ab2b0b9f964a520e56b20e3-1              0   
2  e-4-5a17736b446ea613107b50ce-2              0   
3  e-4-4c3a8e78be4620a1c6180b13-3              0   
4  e-4-549dd93f498eec33464a5fd2-4              0   

                                       reasons.items  \
0  [{'summary': 'This spot is popular', 'type': '...   
1  [{'summary': 'This spot is popular', 'type': '...   
2  [{'summary': 'This spot is popular', 'type': '...   
3  [{'summary': 'This spot is popular', 'type': '...   
4  [{'summary': 'This spot is popular', 'type': '...   

                   venue.id                         venue.name  \
0  4ad7bf71f964a5205b0e21e3                 Sin and Redemption   
1  4ab2b0b9f964a520e56b20e3  The Queen and Beaver Public House   
2  5a17736b446ea613107b50ce                          Suite 114   
3  4c3a8e78be4620a1c6180b13    Hilton Toronto Executive Lounge   
4  549dd93f498eec33464a5fd2                Queen St. Warehouse   

     venue.location.address venue.location.crossStreet  venue.location.lat  \
0            136 McCaul St.              at Dundas St.           43.654661   
1                35 Elm St.                 at Bay St.           43.657472   
2           116 Dundas St W                        NaN           43.655766   
3  145 Richmond Street West                        NaN           43.650363   
4            232 Queen St W                    John St           43.650117   

   venue.location.lng                      venue.location.labeledLatLngs  \
0          -79.391737  [{'label': 'display', 'lat': 43.65466135819523...   
1          -79.383524  [{'label': 'display', 'lat': 43.65747228208784...   
2          -79.384310  [{'label': 'display', 'lat': 43.655766, 'lng':...   
3          -79.385802  [{'label': 'display', 'lat': 43.65036312586515...   
4          -79.390316  [{'label': 'display', 'lat': 43.65011654059042...   

   venue.location.distance venue.location.postalCode venue.location.cc  \
0                      373                   M5T 1W2                CA   
1                      490                   M5G 1H1                CA   
2                      307                   M5G 1C3                CA   
3                      416                   M5G 1Y1                CA   
4                      496                   M5V 1Z6                CA   

  venue.location.city venue.location.state venue.location.country  \
0             Toronto                   ON                 Canada   
1             Toronto                   ON                 Canada   
2             Toronto                   ON                 Canada   
3             Toronto                   ON                 Canada   
4             Toronto                   ON                 Canada   

                     venue.location.formattedAddress  \
0  [136 McCaul St. (at Dundas St.), Toronto ON M5...   
1  [35 Elm St. (at Bay St.), Toronto ON M5G 1H1, ...   
2      [116 Dundas St W, Toronto ON M5G 1C3, Canada]   
3  [145 Richmond Street West, Toronto ON M5G 1Y1,...   
4  [232 Queen St W (John St), Toronto ON M5V 1Z6,...   

                                    venue.categories  venue.photos.count  \
0  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...                   0   
1  [{'id': '4bf58dd8d48988d155941735', 'name': 'G...                   0   
2  [{'id': '4bf58dd8d48988d1d4941735', 'name': 'S...                   0   
3  [{'id': '4bf58dd8d48988d143941735', 'name': 'B...                   0   
4  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...                   0   

  venue.photos.groups  
0                  []  
1                  []  
2                  []  
3                  []  
4                  []

In [26]:
list(toronto_drinks_df.columns) 

['referralId',
 'reasons.count',
 'reasons.items',
 'venue.id',
 'venue.name',
 'venue.location.address',
 'venue.location.crossStreet',
 'venue.location.lat',
 'venue.location.lng',
 'venue.location.labeledLatLngs',
 'venue.location.distance',
 'venue.location.postalCode',
 'venue.location.cc',
 'venue.location.city',
 'venue.location.state',
 'venue.location.country',
 'venue.location.formattedAddress',
 'venue.categories',
 'venue.photos.count',
 'venue.photos.groups']

In [27]:
toronto_drinks2_df = toronto_drinks_df[['venue.id','venue.name','venue.location.lat', 'venue.location.lng', 'venue.location.formattedAddress']]
toronto_drinks2_df

venue.id                         venue.name  \
0  4ad7bf71f964a5205b0e21e3                 Sin and Redemption   
1  4ab2b0b9f964a520e56b20e3  The Queen and Beaver Public House   
2  5a17736b446ea613107b50ce                          Suite 114   
3  4c3a8e78be4620a1c6180b13    Hilton Toronto Executive Lounge   
4  549dd93f498eec33464a5fd2                Queen St. Warehouse   

   venue.location.lat  venue.location.lng  \
0           43.654661          -79.391737   
1           43.657472          -79.383524   
2           43.655766          -79.384310   
3           43.650363          -79.385802   
4           43.650117          -79.390316   

                     venue.location.formattedAddress  
0  [136 McCaul St. (at Dundas St.), Toronto ON M5...  
1  [35 Elm St. (at Bay St.), Toronto ON M5G 1H1, ...  
2      [116 Dundas St W, Toronto ON M5G 1C3, Canada]  
3  [145 Richmond Street West, Toronto ON M5G 1Y1,...  
4  [232 Queen St W (John St), Toronto ON M5V 1Z6,...

In [28]:
toronto_venues_id_list = toronto_drinks2_df['venue.id'].values.tolist()
toronto_venues_id_list

['4ad7bf71f964a5205b0e21e3',
 '4ab2b0b9f964a520e56b20e3',
 '5a17736b446ea613107b50ce',
 '4c3a8e78be4620a1c6180b13',
 '549dd93f498eec33464a5fd2']

In [29]:
toronto_rating = []
toronto_price = []
toronto_likes = []

for toronto_venues1 in toronto_venues_id_list:
    url_t2 = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(toronto_venues1, CLIENT_ID, CLIENT_SECRET, VERSION)
    result_t2 = requests.get(url_t2).json()
    result_t4 = result_t2['response']['venue']['rating']
    result_t5 = result_t2['response']['venue']['attributes']['groups'][0]['summary']
    result_t6 = result_t2['response']['venue']['likes']['count']
    toronto_rating.append(result_t4)
    toronto_price.append(result_t5)
    toronto_likes.append(result_t6)

toronto_rpl = pd.DataFrame({'Price': toronto_price, 'Rating': toronto_rating, 'Number of likes': toronto_likes})
toronto_rpl

Price  Rating  Number of likes
0    $$     8.0              153
1    $$     8.5              233
2   $$$     7.5                6
3   $$$     7.8               29
4    $$     7.9              122

In [30]:
toronto_v1 = pd.concat([toronto_drinks2_df, toronto_rpl], axis=1)
toronto_v1

venue.id                         venue.name  \
0  4ad7bf71f964a5205b0e21e3                 Sin and Redemption   
1  4ab2b0b9f964a520e56b20e3  The Queen and Beaver Public House   
2  5a17736b446ea613107b50ce                          Suite 114   
3  4c3a8e78be4620a1c6180b13    Hilton Toronto Executive Lounge   
4  549dd93f498eec33464a5fd2                Queen St. Warehouse   

   venue.location.lat  venue.location.lng  \
0           43.654661          -79.391737   
1           43.657472          -79.383524   
2           43.655766          -79.384310   
3           43.650363          -79.385802   
4           43.650117          -79.390316   

                     venue.location.formattedAddress Price  Rating  \
0  [136 McCaul St. (at Dundas St.), Toronto ON M5...    $$     8.0   
1  [35 Elm St. (at Bay St.), Toronto ON M5G 1H1, ...    $$     8.5   
2      [116 Dundas St W, Toronto ON M5G 1C3, Canada]   $$$     7.5   
3  [145 Richmond Street West, Toronto ON M5G 1Y1,...   $$$     7.8   
4  [232 Queen St W (John St), Toronto ON M5V 1Z6,...    $$     7.9   

   Number of likes  
0              153  
1              233  
2                6  
3               29  
4              122

In [31]:
toronto_v2 = toronto_v1.rename(columns={"venue.id": "ID", "venue.name": "Venue", "venue.location.lat": "Latitude", "venue.location.lng": "Longitude", "venue.location.formattedAddress": "Address"})

In [32]:
toronto_v3 = toronto_v2.replace({'$$': '$10-$20', '$$$':'$20-$30'})

<BR><BR><B><font color = 'blue'> Data </font><a name="data"></B> </H3>

<B> NEW YORK CITY </B>

In [35]:
nyc_v3

ID                         Venue   Latitude  \
0  57f0689d498e7d49d9189369  The Bar Room at Temple Court  40.711448   
1  534c9d7b498e1bdd443a40e1                       Racines  40.714754   
2  4db5fbfe43a1369cb5dc2a1f    Maxwell's Bar & Restaurant  40.714944   
3  4aa0669cf964a5207e3f20e3                     The Wooly  40.712212   
4  433b2e80f964a52036281fe3         FlashDancers Downtown  40.714350   

   Longitude                                            Address  \
0 -74.006802  [123 Nassau St, New York, NY 10038, United Sta...   
1 -74.007581  [94 Chambers St, New York, NY 10007, United St...   
2 -74.006485  [59 Reade St (at Broadway), New York, NY 10007...   
3 -74.008357  [11 Barclay St (at Broadway), New York, NY 102...   
4 -74.009800  [59 Murray St (Church St & West Broadway), New...   

                                        Delivery url    Price  Rating  \
0                                      Not available  $20-$30     9.4   
1                                      Not available  $20-$30     8.6   
2  https://www.seamless.com/menu/maxwells-59-read...  $10-$20     8.1   
3  https://www.grubhub.com/restaurant/the-wooly-p...  $20-$30     7.8   
4                                      Not available  $20-$30     8.7   

   Number of likes  
0              182  
1              115  
2              165  
3              173  
4               87

In [39]:
nyc_map = folium.Map(location=[latitude, longitude], zoom_start=15)

for lat, lng in zip(nyc_v3.Latitude, nyc_v3.Longitude):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(nyc_map)

nyc_map

<B> TORONTO </B>

In [36]:
toronto_v3

ID                              Venue   Latitude  \
0  4ad7bf71f964a5205b0e21e3                 Sin and Redemption  43.654661   
1  4ab2b0b9f964a520e56b20e3  The Queen and Beaver Public House  43.657472   
2  5a17736b446ea613107b50ce                          Suite 114  43.655766   
3  4c3a8e78be4620a1c6180b13    Hilton Toronto Executive Lounge  43.650363   
4  549dd93f498eec33464a5fd2                Queen St. Warehouse  43.650117   

   Longitude                                            Address    Price  \
0 -79.391737  [136 McCaul St. (at Dundas St.), Toronto ON M5...  $10-$20   
1 -79.383524  [35 Elm St. (at Bay St.), Toronto ON M5G 1H1, ...  $10-$20   
2 -79.384310      [116 Dundas St W, Toronto ON M5G 1C3, Canada]  $20-$30   
3 -79.385802  [145 Richmond Street West, Toronto ON M5G 1Y1,...  $20-$30   
4 -79.390316  [232 Queen St W (John St), Toronto ON M5V 1Z6,...  $10-$20   

   Rating  Number of likes  
0     8.0              153  
1     8.5              233  
2     7.5                6  
3     7.8               29  
4     7.9              122

In [41]:
toronto_map = folium.Map(location=[43.653963, -79.387207], zoom_start=15)

for lat, lng in zip(toronto_v3.Latitude, toronto_v3.Longitude):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='red',
        fill = True,
        fill_color='red',
        fill_opacity=0.6
    ).add_to(toronto_map)

toronto_map

<BR><BR><B><font color = 'blue'> Results and Discussion </font><a name="discussion"></B>

Our data shows that NYC has higher ratings and also pricier.

The rating for NYC range from 7.8 to 9.4. Only one of the five venue is tier price 2 (\\$10 - \\$20) and the rest are tier price 3 (\\$20 - \\$30). Although Flash Dancers Downtown has the second highest rating of 8.7, note that there are only 87 likes. Interestingly, two of the venues have delivery, however it is most likely related to food and not drinks. 

The rating for Toronto range from 7.5 to 8.0. Three of the venues are tier price 2 and the other two are tier price 3. Interestingly, Suite 114 and Hilton Toronto Executive Lounge only has 6 and 29 likes respectively. The other three venues have more likes, indicating a better data population.

<U> Analysis </U><BR>
Overall, it is difficult to make the comparison between drinking venues in NYC and Toronto just based on price, ratings and likes. The reasons include the following:
* The number of customers is different, therefore the data population may be skewed. 
* Some venues seem to offer both drinks and food - which makes it difficult to understand if it is the food that was rated high or the drinks. 
* Different venue offer different menu, a deeper look into the menu may offer more valuable insights
* Further scraping on the comments/reviews per venue may offer more insights into customer taste
    
In addition, I find it interesting how Foursquare have sorted the venues popularity. While the url was set to 'sortByPopularity=0', the data generated did not sort the ratings from high to low. The sort by popularity could also be set to 1, which I tried as an experiment and it yielded lower number of likes therefore I decided to keep it at zero. Understanding how Foursquare sorted the venues would be crucial to have better confidence if the venues generated were indeed the top venues.

<BR><BR><B><font color = 'blue'> Conclusion </font><a name="conclusion"></B>

While the data was inconclusive, we can agree that NYC is more popular due to more number of likes. More data scraping would be required to further determine the differences for customers to determine which city would be a better fit for them.